toast-tk-plugin-loader/src/main/java/io/toast/tk/plugin/AgentBoot.java

```diff
@@ -0,0 +1,42 @@
+package io.toast.tk.plugin;
+
+import java.lang.instrument.Instrumentation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.toast.tk.agent.config.AgentConfigProvider;
+
+public class AgentBoot {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AgentBoot.class);
+
+	private static Instrumentation instrumentation;
+
+	/**
+	 * JVM hook to statically load the javaagent at startup.
+	 * 
+	 * After the Java Virtual Machine (JVM) has initialized, the premain method
+	 * will be called. Then the real application main method will be called.
+	 * 
+	 */
+	public static void premain(
+		String args,
+		Instrumentation inst)
+		throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.Exception', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)